src/dependency.rs

```diff
@@ -29,8 +29,18 @@ impl FromStr for Dependency {
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         static RE: LazyLock<Regex> = LazyLock::new(|| {
-            Regex::new(r"^(?:(?P<url>(?:http|https|ssh)://(?:[^:@]+(?::[^@]+)?@)?[^/:@?#+=~]+(?:/[^#+=~]+)+(?:\.git)?)|(?P<name>[^=+]+))(?:#(?P<git_ref>[^+=]+))?(?:=(?P<version>(?:>=|<=|>|<|=|~)?[0-9A-Za-z\.\-]+))?(?P<default_features>\+[^+]?)?(?P<features>(?:\+[^+]+)*)$")
-            .expect("dependency's regex must be compiled")
+            let pattern = concat!(
```

Comment:
When regex get too complicated we usually activate the flag x which is the verbose mode https://docs.rs/regex/1.12.2/regex/#grouping-and-flags

That flag is really meant for forcing readability of hard regex:
- explicit white spaces are now enforced and clearly visible to the developer
- you can actually comment your regex with `#`

This is the more conventional way to handle this particular issue.